main.js

```diff
@@ -0,0 +1,186 @@
+  currentWeather = [];
+};
+
+//Extracts specific information from the API and pushes it to the forecastWeather array
+var addForecast = function (data) {
+  //Restructures the data to be grouped by day
+  var weatherByDate = groupForecast(data.list);
+
+//Calls each of the four functions to create arrays for forecast data
+ var avgTemps = findAvgTemp(weatherByDate);
+ var forecastDays = findForecastDays(weatherByDate);
+ var conditions = getConditions(weatherByDate);
+ var iconIds = getIconIds(weatherByDate);
+ 
+ for(i=0; i < Object.keys(weatherByDate).length - 1; i++) {
+   forecastWeather.push({
+     day: forecastDays[i],
+     weatherType: conditions[i],
+     degrees: avgTemps[i] + '°',
+     iconURL: 'https://openweathermap.org/img/wn/' + iconIds[i] + '@2x.png'
+    });
+  };
+  
+  renderForecast();
+  forecastWeather = [];
+};
+
+var renderWeather = function () {
+  $('.currentWeather').empty();
+
+  for (var i = 0; i < currentWeather.length; i++) {
+    const weather = currentWeather[i];
+
+    // create HTML and append to .currentWeather
+    var source = $('#current-weather-template').html();
+    var template = Handlebars.compile(source);
+    var newHTML = template(weather);
+
+    $('.currentWeather').append(newHTML);
+  }
+};
+
+var renderForecast = function () {
+  $('.forecast').empty();
+
+  for (var i = 0; i < forecastWeather.length; i++) {
+    const weatherForecast = forecastWeather[i];
+    // create HTML and append to .currentWeather
+    var source = $('#forecast-template').html();
+    var template = Handlebars.compile(source);
+    var newHTML = template(weatherForecast);
+
+    $('.forecast').append(newHTML);
+  }
+};
+
+//Function that groups the weather object from the API into objects by day
+var groupForecast = function (array) {
+  return array.reduce(function (acc, obj) {
+    let key = obj['dt_txt'].substring(0,10);
+    if (!acc[key]) {
+      acc[key] = []
+    };
+    acc[key].push(obj)
+    return acc
+  }, {})
+};
+
+//Function to get today's date
+var getTodaysDate = function () {
+  var today = new Date();
+  var dd = String(today.getDate()).padStart(2, '0');
+  var mm = String(today.getMonth() + 1).padStart(2, '0');
+  var yyyy = today.getFullYear();
+  today = mm + '/' + dd + '/' + yyyy;
+  return today;
+};
+
+//Function to find the daily average temperature using the output from the groupForecast function.
+var findAvgTemp = function (weatherByDate) {
+  var averages = [];
+  for (prop in weatherByDate) {
+    var day = weatherByDate[prop];
+    var temps = [];
+    for(let i = 0; i < day.length; i++) {
+      temps.push(day[i].main.temp);
+    };
+    const sum = temps.reduce((accumulator,currentValue) => {
+      return accumulator + currentValue;
+    });
+    var avg = Math.round(sum/temps.length);
+    averages.push(avg);
+  };
+  return averages;
+};
+
+//function that returns an array of days to use in five day forecast
+var findForecastDays = function (weatherByDate) {
+  getTodaysDate();
```

Comment:
You should avoid doing this - instead, write something like:

```js
var today = getToday();
```

Again, you're creating a side effect b/c `getToday` mutating a global variable. You're right to return `today` from `getToday`, but you're not utilizing that returned value.